CAKE-5292 | Use only one call to the service AND OTHER HOTFIXES

vforge opened the pull request:
## Links

* http://wikia-inc.atlassian.net/browse/CAKE-5292

## Description

- Rework how `affiliate-slot` service works, always call service once and always return `{big: unit, small: unit}` object from Promise
- other hotfixes

## Reviewers

@Wikia/cake

vforge committed:
CAKE-5279 | [Nov 18] Enable affiliate units in Australia and New Zealand (with US copy)

vforge committed:
CAKE-5292 | Use only one call to the service

vforge committed:
CAKE-5299 | Only client-side render for interrupt

Commit:
Third debug param

cschucode commented on app/components/article-content.js:
```diff
@@ -605,92 +604,76 @@ export default Component.extend(
+              this.renderedComponents.push(this.renderComponent({
+                name: 'affiliate-unit',
+                attrs: {
+                  unit: bigUnit,
```
Just a question, but were setting the placeholder's unit attribute to always `undefined` here?

vforge replied:
No, why?

JoshuaRogan replied:
I think you are reading the wrong if statement.

cschucode replied:
See it now. if inside the if. Thanks. 👍

Commit:
Add correct aff link

vforge committed:
lint and fixes

vforge committed:
Merge branch 'cake-5292' into cake-5299

vforge committed:
CAKE-5299 | Only client-side render for interrupt (#1914)

CAKE-5299 | Only client-side render for interrupt

JoshuaRogan commented:
New debug feat: 
`debugAffiliateUnits=disneyplus,disney,true` the third param is ignoreBigUnit which can be ignored or specified to only display a small unit.

vforge committed:
CAKE-5279 | [Nov 18] Enable affiliate units in Australia and Ne… (#1909)

CAKE-5279 | [Nov 18] Enable affiliate units in Australia and New Zealand (with US copy)

mklucsarits commented on app/services/affiliate-slots-units.json:
```diff
@@ -112,5 +112,167 @@
+    "image": "https://static.wikia.nocookie.net/d8374c50-15b4-4ee6-97be-fa8dc4d5976e",
+    "heading": "All Avengers, All the Time",
+    "subheading": "Watch Now On Disney+",
+    "link": "https://disneyplus.bn5x.net/beeqP",
```
This is the old link

JoshuaRogan replied:
https://github.com/Wikia/mobile-wiki/pull/1913/commits/ab61b542efc808b2bd2768eeef6c29dce3977dbe

mklucsarits commented on app/services/affiliate-slots-units.json:
```diff
@@ -112,5 +112,167 @@
+    "image": "https://static.wikia.nocookie.net/d8374c50-15b4-4ee6-97be-fa8dc4d5976e",
+    "heading": "All Avengers, All the Time",
+    "subheading": "Watch Now On Disney+",
+    "link": "https://disneyplus.bn5x.net/beeqP",
```
This also needs to be updated

JoshuaRogan replied:
Done https://github.com/Wikia/mobile-wiki/pull/1913/commits/ab61b542efc808b2bd2768eeef6c29dce3977dbe

Commit:
Update json files
